aiida/orm/nodes/node.py

```diff
@@ -180,6 +180,17 @@ def __init__(self, backend=None, user=None, computer=None, **kwargs):
         )
         super().__init__(backend_entity)
 
+    def __eq__(self, other):
+        """Fallback equality comparison by uuid (can be overwritten by specific types)"""
+        if isinstance(other, Node) and self.uuid == other.uuid:
+            return True
+        return super().__eq__(other)
+
+    def __hash__(self):
+        """Python-Hash: Implementation that is compatible with __eq__"""
+        from uuid import UUID
```

Comment:
Since that's a standard library import, I think it can go at the top of the file.